base/linalg/uniformscaling.jl

```diff
@@ -171,6 +171,15 @@ function isapprox(J1::UniformScaling{T}, J2::UniformScaling{S};
             rtol::Real=Base.rtoldefault(T,S), atol::Real=0, nans::Bool=false) where {T<:Number,S<:Number}
     isapprox(J1.λ, J2.λ, rtol=rtol, atol=atol, nans=nans)
 end
+function isapprox(J::UniformScaling,A::AbstractMatrix;kwargs...)
+    n = checksquare(A)
+    D = diagm(J.λ*ones(typeof(J.λ),n))
+    return isapprox(D,A;kwargs...)
+end
```

Comment:
The version I suggested is significantly more efficient.  This one constructs two temporary square matrices (`D` and `A-D` rather than one as in my suggestion, and computes `A-D` the slow way (subtracting every element rather than just the diagonals).  Why not do it my way?